### PR TITLE
meson: add patch to find boost via pkg-config

### LINKED ISSUE #86131: Meson no longer able to pick up Boost
(originally about `pulseeffects` build)

Broken on Hydra for a few days now; going by [the log](https://hydra.nixos.org/build/117384596/nixlog/1) it seems to be a boost issue:

```
Run-time dependency Boost (missing: filesystem, system) found: NO 

src/meson.build:97:0: ERROR: Dependency "boost" not found
```

but removing the boost version override introduced in 470b916556a800147fce11aef05fc503663cbdba didn't fix it (and https://github.com/wwmm/pulseeffects/issues/645 suggests it's required anyway). Unfortunately I'm not very familiar with boost or meson so I'm not sure what to poke at next, but I'll update this issue if I figure anything out.

cc @jtojnar @matthiasbeyer

### PR DESCRIPTION
**Pull request by Artturin**
Fixes https://github.com/NixOS/nixpkgs/issues/86131

https://github.com/mesonbuild/meson/pull/13272

TODO remove all the meson(cmake uses the vars too) `BOOST_{INCLUDE,LIBRARY}DIR`

Tested with

```diff
diff --git a/pkgs/tools/package-management/lix/common.nix b/pkgs/tools/package-management/lix/common.nix
index fe77830d56f1..f25b16af3309 100644
--- a/pkgs/tools/package-management/lix/common.nix
+++ b/pkgs/tools/package-management/lix/common.nix
@@ -21,6 +21,7 @@ assert (hash == null) -> (src != null);
   bash,
   bison,
   boehmgc,
+  fetchpatch,
   boost,
   brotli,
   busybox-sandbox-shell,
@@ -110,7 +111,16 @@ let
         bison
         flex
         jq
-        meson
+        (meson.overrideAttrs (previousAttrs: {
+          patches = previousAttrs.patches ++ [
+            # upstream PR: https://github.com/mesonbuild/meson/pull/13272
+            (fetchpatch {
+              name = "find-boost-pkg-config.patch";
+              url = "https://github.com/mesonbuild/meson/commit/f0eda0346b2b936e4ef795f3a82733d6711b18fb.patch";
+              sha256 = "sha256-uSilNuSx9yd1cxs0XVLcLw4MOXEd2uIe2g+wk+SBqeU=";
+            })
+          ];
+        }))
         ninja
         cmake
         python3
@@ -201,13 +211,6 @@ let
         (lib.mesonOption "sandbox-shell" "${busybox-sandbox-shell}/bin/busybox")
       ];
 
-    # Needed for Meson to find Boost.
-    # https://github.com/NixOS/nixpkgs/issues/86131.
-    env = {
-      BOOST_INCLUDEDIR = "${lib.getDev boost}/include";
-      BOOST_LIBRARYDIR = "${lib.getLib boost}/lib";
-    };
-
     postInstall =
       ''
         mkdir -p $doc/nix-support
```

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/me/meson/package.nix [hunk 1]
```diff
@@ -63,6 +64,14 @@ python3.pkgs.buildPythonApplication rec {
     # https://github.com/NixOS/nixpkgs/issues/86131#issuecomment-711051774
     ./005-boost-Do-not-add-system-paths-on-nix.patch
 
+    # https://github.com/NixOS/nixpkgs/issues/86131
+    # upstream PR: https://github.com/mesonbuild/meson/pull/13272
+    (fetchpatch {
+      name = "find-boost-pkg-config.patch";
+      url = "https://github.com/mesonbuild/meson/commit/f0eda0346b2b936e4ef795f3a82733d6711b18fb.patch";
```
Comment (alyssais) [inline pkgs/by-name/me/meson/package.nix:71]: URLs for unmerged commits are not stable — if you rebase the patch this commit will be garbage collected and not exist, so for longevity this sort of patch needs to be vendored.

**Review by alyssais — CHANGES_REQUESTED** [review]

**Comment by Artturin** [comment]
The upstream PR was merged.

**Review by Lurkki14 — APPROVED** [review]

**Comment by Artturin** [comment]
Rebased

**Comment by alyssais** [comment]
Thanks for getting this fixed upstream!

**Pull request merged**